### DISCUSSION TITLE
Remote agent workers

### DISCUSSION THREAD

**Discussion by itsh3sam** [windmill-labs/windmill#3093] [Questions] [answered]
Is it possible to create agent workers without the need of main database access?

**Comment by rubenfiszel** [comment, marked as answer]
Currently not but it is one of the improvement we are exploring with a large enterprise customer for high latency agents. We can prioritize this based on demand but it will be an EE feature.

**Comment by rubenfiszel** [comment]
@AnonC0DER  I'm afraid my answer was imprecise. It is possible to create agent workers without the "main" database access. The required roles are defined here: https://www.windmill.dev/docs/core_concepts/agent_workers

What we are working on is for the agent workers to only interact with HTTP to the servers and not require any database access. This will be done in the next following months.

**Reply by itsh3sam** [reply]
Thanks
